### PR TITLE
7483 Skip delivery address if empty

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#7483

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/__init__.py [hunk 1]
```diff
@@ -107,8 +107,9 @@ def create_party(business_id: int, party_info: dict, create: bool = True) -> Par
         )
 
     # add addresses to party
-    address = create_address(party_info.get('deliveryAddress', None), Address.DELIVERY)
```
Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/__init__.py:110]: This creates an empty Address in address and address_version table

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/__init__.py [hunk 2]
```diff
@@ -107,8 +107,9 @@ def create_party(business_id: int, party_info: dict, create: bool = True) -> Par
...
-    party.delivery_address = address
+    if party_info.get('deliveryAddress', None):
```
Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/__init__.py:110]: Delivery address will not be present for 'Completing Party' and 'Incorporator'

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/__init__.py [hunk 3]
```diff
@@ -107,8 +107,9 @@ def create_party(business_id: int, party_info: dict, create: bool = True) -> Par
         )
 
     # add addresses to party
-    address = create_address(party_info.get('deliveryAddress', None), Address.DELIVERY)
-    party.delivery_address = address
```
Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/__init__.py:111]: This creates an empty Address in address and address_version table

File: legal-api/src/legal_api/services/business_details_version.py [hunk 4]
```diff
@@ -459,12 +459,15 @@ def party_revision_json(transaction_id, party_revision, is_ia_or_after) -> dict:
...
-                VersionedBusinessDetailsService.get_address_revision
-                (transaction_id, party_revision.delivery_address_id))
-            if 'addressType' in member_address:
-                del member_address['addressType']
-            member['deliveryAddress'] = member_address
+            address_revision = VersionedBusinessDetailsService.get_address_revision(
+                transaction_id, party_revision.delivery_address_id)
+            # This condition can be removed once we correct data in address and address_version table
+            # by removing empty address entry.
+            if address_revision and address_revision.postal_code:
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/business_details_version.py:466]: This condition is required until we fix the existing data

**Review by andrepestana-aot — APPROVED** [review]
LGTM!

**Review by kialj876 — APPROVED** [review]

**Pull request merged**